src/app/[id]/page.tsx

```diff
@@ -0,0 +1,160 @@
+"use server";
+
+import React from "react";
+import { Box, Divider, Typography } from "@mui/material";
+
+import Rating from "@/components/ui/Stars/rating";
+import ThemeWrapper from "@/components/ThemeWrapper";
+import AddToCart from "@/components/id/AddToCart/AddToCard";
+
+import getProducts from "@/utils/getProducts";
+import calculateDiscountedPrice from "@/utils/calculateDiscountedPrice";
+import { ProductType } from "@/types/product";
+
+import styles from "./page.styles";
+
+const ImageCarousel = React.lazy(
+  () => import("@/components/id/ImageCarousel/ImageCarousel")
+);
+const ProductsDetails = React.lazy(
+  () => import("@/components/id/ProductsDetails/ProductsDetails")
+);
+const Reviews = React.lazy(() => import("@/components/id/Reviews/Reviews"));
+
+const ProductPage = async ({ params }: { params: { id: number } }) => {
+  const product = await getProducts(params.id);
+
+  if (!product || Array.isArray(product)) {
+    return (
+      <Box component="main" sx={styles.main}>
+        <Typography variant="body1" color="error">
+          Failed to load product details.
+        </Typography>
+      </Box>
+    );
+  }
+
+  const discountPrice = product.discountPercentage
+    ? calculateDiscountedPrice(product.price, product.discountPercentage)
+    : undefined;
+
+  return (
+    <ThemeWrapper>
+      <MainSection {...product} discountPrice={discountPrice} />
+    </ThemeWrapper>
+  );
+};
+
+export const generateStaticParams = async () => {
+  try {
+    const response = await fetch("https://dummyjson.com/products");
+    const data = await response.json();
```

Comment:
Also why u didn't use a function from the api u have written, diffrent level of [abstraction](https://medium.com/@danieljyoo/levels-of-abstraction-a-key-concept-in-systems-design-7fdb33d288af)